File: library/src/main/java/xyz/fcampbell/rxgms/observables/GoogleAPIConnectionSuspendedException.java
package xyz.fcampbell.rxgms.observables;

public class GoogleAPIConnectionSuspendedException extends RuntimeException {
    private final int cause;

    GoogleAPIConnectionSuspendedException(int cause) {
        this.cause = cause;
    }

    public int getErrorCause() {
        return cause;
    }
}

File: library/src/main/kotlin/xyz/fcampbell/rxplayservices/common/exception/GoogleApiConnectionSuspendedException.kt
package xyz.fcampbell.rxplayservices.common.exception

class GoogleApiConnectionSuspendedException : RuntimeException {
    val errorCause: Int

    internal constructor(errorCause: Int) : super() {
        this.errorCause = errorCause
    }

    internal constructor(errorCause: Int, detailMessage: String) : super(detailMessage) {
        this.errorCause = errorCause
    }

    internal constructor(errorCause: Int, cause: Throwable) : super(cause) {
        this.errorCause = errorCause
    }

    internal constructor(errorCause: Int, detailMessage: String, cause: Throwable) : super(detailMessage, cause) {
        this.errorCause = errorCause
    }
}
